workspaces/ballerina/ballerina-visualizer/src/views/BI/Forms/FormGeneratorNew/index.tsx

```diff
@@ -818,30 +822,30 @@ export function FormGeneratorNew(props: FormProps) {
                     title="Create New Type"
                     openState={typeEditorState.isOpen}
                     setOpenState={handleTypeEditorStateChange}>
-                    <div style={{ padding: '0px 15px' }}>
-                        {stack.slice(0, i + 1).slice(0, i + 1).length > 1 && (
+                    <div style={{ padding: '0px 20px' }}>
+                        {stack.slice(0, i + 1).length > 1 && (
                             <BreadcrumbContainer>
                                 {stack.slice(0, i + 1).map((stackItem, index) => (
                                     <React.Fragment key={index}>
                                         {index > 0 && <BreadcrumbSeparator>/</BreadcrumbSeparator>}
                                         <BreadcrumbItem>
-                                            {stackItem?.type?.name || "New Type"}
+                                            {stackItem?.type?.name || "NewType"}
                                         </BreadcrumbItem>
                                     </React.Fragment>
                                 ))}
                             </BreadcrumbContainer>
                         )}
                         <FormTypeEditor
-                            type={ isGraphqlEditor? defaultType() : undefined}
+                            type={peekTypeStack()?.type}
```

Comment:
Here we need to check if its a graphqlEditor. based on that we render the matching default type. Records for params and ServiceClass for return types